backend-engineer/importer/README.MD

```diff
@@ -0,0 +1,109 @@
+
+## ✨ Overview:
+
+Design and implement a robust data ingestion and querying system using a modular backend architecture. 
+The goal is to simulate a real-world import pipeline that scales well and meets performance, quality, and maintainability standards.
+
+Please read carefully all the instructions. If you feel that something is not clear, please reach out, or feel free to make an educated guess!
+
+
+## 📆 Time estimate
+
+Please submit your ideas within 7 days of receiving the challenge, so that we have enough time to review your solution with the rest of the team
+before the next interview. 
+
+Based on our previous experience, and the available tooling nowadays, we believe that it should take no more than 8-10h.
+But do not let this to be a limiting factor to you. Good enough means good enough, but it's up to you how much do you want to polish it
+
+---
+
+## 1. Requirements
+
+You are required to implement a small system that allows importing and querying environmental data.
+
+This system must include:
+
+### A. **CSV Import Microservice**
+
+* Accepts a CSV upload (e.g., via POST endpoint).
+* Parses and validates the file.
+* Stores the data in a **relational database**.
+* Performs basic aggregation (e.g., record count, min/max for numeric fields) post-import.
+
+### B. **Query API Microservice**
+
+* Exposes a REST API to query the imported dataset.
+* Must support:
+    * Filtering by any field (query params)
+    * Pagination and sorting
+* Must return results quickly.
+
+### C. **Infrastructure & Tooling**
+
+* Use a single **public** GitHub repository (no need to create separate repositories for the services)
+* Provide a comprehensive commit tree
+* Use **Docker + docker-compose** to deliver a working setup. It 
+* Provide **unit and/or integration tests**.
+* Use Typescript and any Node.js framework and ORM (e.g., Drizzle, Prisma, TypeORM).
+* Provide any code-styling tool of your choice (ESLint)
+* Add a basic testing pipeline to your project. Your code should be automatically tested when pushed upstream.
+* Document the setup and usage in a `README.md`.
+
+
+---
+
+## 2. Optional goals
+
+Found the basic requirements too easy? Want to go the extra mile to impress us? 
+Whatever the reason is, here are some optional ways in which you can give us a better sense of your approach, of your analytical and communication skills and of what you care about:
+
+* Use **two different languages/frameworks** (e.g., NodeJS for import, Python FastAPI for query).
+* Add a basic **rate limiter** to the query service.
+* Include profiling results or comments on performance considerations.
+* Add a metadata endpoint (e.g., `/status`) to expose info like total records, last import, schema version.
+* We use and love open source software, and all OSS code must be properly documented. Add relevant documentation, beyond the “quickstart”, as you would for a publicly available OSS project.
+* Do you think you can add useful features, or 
```

Comment:
I sense a dramatic turn of events may be next, but the spoiler leaves me dangling... or... what? 😛

Reply:
lol, I must've short circuited. updated